### PR TITLE
Make server configurable

### PR DESCRIPTION
**Pull request by PGijsbers**
This is some work that makes the server configurable at runtime. Usage:

```python
import aiod
from aiod.config import config

aiod.get_list()  # reaches the default endpoint (production: "https://api.aiod.eu/")

config.api_base_url = "https://api.aiod.eu/"
aiod.get_list()  # reaches the new endpoint (http://localhost/)
```

The previous design used constants. In some ways this is great: no changing global mutable state to keep track of. However, this also makes changing the desired endpoint at runtime impossible. If we keep that design, we would for example need to add parameters to each function call so allow changing of the endpoint.

Alternatively, we could simple embrace that design and enforce that endpoints can not be changed at runtime. The main objection to that is for me that in the future scripts may alternate between different endpoints (e.g., the root nodes and specific leaf nodes).

*note:* I personally don't like the double config in `from aiod.config import config`. We can make some [slight modifications](https://github.com/aiondemand/aiondemand/compare/feat/configurable-server...feat/configurable-server-alt) and then simply have `aiod.config` work. It used a little bit of a hack, but I think it's not too hard to understand (and is transparent to users).

### CODE REVIEW

**Comment by jsmatias** [comment]
Weirdly, Github is comparing the files to a previous version of the `develop` branch. I marked it ready to review and put it back to draft, but it didn't work. Anyway, there were not many changes, I can still compare them as it is.

File: src/aiod/__init__.py [hunk 1]
```diff
@@ -17,5 +16,3 @@
 from aiod.resources import publications
 from aiod.resources import services
 from aiod.resources import teams
```
Comment (jsmatias) [inline src/aiod/__init__.py:18]: Regarding the `from aiod import config`. Why not simply import it here in the `__init__.py` file?
```suggestion
from aiod.resources import teams
from aiod.config._config import config
```

**Review by jsmatias — COMMENTED** [review]
I think it's good enough for now. I just left one suggestion.

File: src/aiod/__init__.py (hunk 1 above)
Comment (PGijsbers) [reply src/aiod/__init__.py:18]: A third thing we could do is for the `calls/urls.py` and others to not import it from `aiod` but instead directly from `aiod.config._config`. Might be better.

Comment (jsmatias) [reply src/aiod/__init__.py:18]: Great, I'll push your suggestions in a minute with a tiny modification:
1. Rename `config` module to `configuration`
2. Include `from _config import config` at `configuration/__init__.py`
3. Include `from aiod.configuration import config` at `aiod.__init__.py`
4. Import `config` from `aiod` directly from `aiod.configuration` in the files such as `calls/urls.py`

Looks cleaner to me.

Comment (PGijsbers) [reply src/aiod/__init__.py:18]: Agreed. Great!

**Review by jsmatias — APPROVED** [review]
Github is complaining about some conflicts, but it seems to be comparing this branch with old files, not the ones in the current version of the develop branch. 

Otherwise, it looks good to me.

**Review by jsmatias — APPROVED** [review]
All solved now. ✅

**Pull request merged**